config/config-api/src/main/java/com/thoughtworks/go/config/PipelineConfig.java

```diff
@@ -560,35 +560,20 @@ public boolean isFirstStageManualApproval() {
 
     @Override
     public boolean equals(Object o) {
-        if (this == o) {
-            return true;
-        }
-        if (o == null || getClass() != o.getClass()) {
-            return false;
-        }
-        if (!super.equals(o)) {
-            return false;
-        }
-
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        if (!super.equals(o)) return false;
         PipelineConfig that = (PipelineConfig) o;
-
-        if (labelTemplate != null ? !labelTemplate.equals(that.labelTemplate) : that.labelTemplate != null) {
-            return false;
-        }
-        if (materialConfigs != null ? !materialConfigs.equals(that.materialConfigs) : that.materialConfigs != null) {
-            return false;
-        }
-        if (name != null ? !name.equals(that.name) : that.name != null) {
-            return false;
-        }
-        if (timer != null ? !timer.equals(that.timer) : that.timer != null) {
-            return false;
-        }
-        if (trackingTool != null ? !trackingTool.equals(that.trackingTool) : that.trackingTool != null) {
-            return false;
-        }
-
-        return true;
+        return Objects.equals(name, that.name) &&
+                Objects.equals(labelTemplate, that.labelTemplate) &&
+                Objects.equals(params, that.params) &&
+                Objects.equals(trackingTool, that.trackingTool) &&
+                Objects.equals(mingleConfig, that.mingleConfig) &&
+                Objects.equals(timer, that.timer) &&
+                Objects.equals(variables, that.variables) &&
+                Objects.equals(materialConfigs, that.materialConfigs) &&
+                Objects.equals(lockBehavior, that.lockBehavior) &&
+                Objects.equals(templateName, that.templateName);
```

Comment:
Any suggestions on what I should be looking for?

I added those because they are sub elements in the xml config, and think they should be represented. How should I validate it's not broken?

Reply:
@marques-work - I will spend some more time to see where all this is being used and how

Reply:
The only meaningful usage in our codebase for `PipelineConfig.equals()` is in `PipelineConfig.validStagesForFetchArtifact()`. This has a test here:

```
    public void shouldReturnListOfStageConfigWhichIsApplicableForFetchArtifact() {
        PipelineConfig superUpstream = PipelineConfigMother.createPipelineConfigWithStages("superUpstream", "s1", "s2", "s3");
        PipelineConfig upstream = PipelineConfigMother.createPipelineConfigWithStages("upstream", "s4", "s5", "s6");
        upstream.addMaterialConfig(new DependencyMaterialConfig(superUpstream.name(), new CaseInsensitiveString("s2")));

        PipelineConfig downstream = PipelineConfigMother.createPipelineConfigWithStages("downstream", "s7");
        downstream.addMaterialConfig(new DependencyMaterialConfig(upstream.name(), new CaseInsensitiveString("s5")));

        List<StageConfig> fetchableStages = upstream.validStagesForFetchArtifact(downstream, new CaseInsensitiveString("s7"));

        assertThat(fetchableStages.size(), is(2));
        assertThat(fetchableStages, hasItem(upstream.get(0)));
        assertThat(fetchableStages, hasItem(upstream.get(1)));
    }
```

The other 2 usages are in the `equals()` of `PipelineConfigDependencyGraph` and its inner class `PipelineConfigQueueEntry`, but these have no explicit usages that I can find.

What other validations can I perform?